Source/BestProjectEver/Private/Character/BPE_BaseCharacter.cpp

```diff
@@ -44,10 +44,13 @@ void ABPE_BaseCharacter::BeginPlay()
 //----------------------------------------------------------------------------------------------------------------------
 void ABPE_BaseCharacter::HandleCharacterDeath(AActor* DamagedActor, AController* InstigatedBy, AActor* DamageCauser)
 {
-	ABPE_GameplayGameMode* GameplayGameMode = GetWorld()->GetAuthGameMode<ABPE_GameplayGameMode>();
-	if(IsValid(GameplayGameMode))
-	{		
-		GameplayGameMode->OnCharacterDeath(InstigatedBy, GetController(), this);
+	if(GetWorld()->GetAuthGameMode<AGameModeBase>()->IsA(ABPE_GameplayGameMode::StaticClass()))
```

Comment:
This check is not necessary. The way you had it before works fine since it will simply return null if it is not a GameplayGameMode. With this new way, you're pretty much asking the game mode if it is a gameplay game mode twice.